utils/UUID.hh

```diff
@@ -206,7 +206,47 @@ inline std::strong_ordering uuid_tri_compare_timeuuid(bytes_view o1, bytes_view
     return res;
 }
 
-}
+template <typename T>
+class UUID_class {
+    int64_t most_sig_bits;
+    int64_t least_sig_bits;
+public:
+    UUID_class() = default;
```

Comment:
What is the intended semantics of a default constructed `UUID_class`? Currently it just has two uninitialized fields and will compare equal/less/more to other instances depending on what garbage that memory happened to contain. I think we need a well defined state for a default constructed `UUID_class`.

Reply:
Right.  The intention is to keep it as uninitialized so it will be trivially default constructible (See https://en.cppreference.com/w/cpp/types/is_trivial)

Reply:
I think you can still initialize members as in `int64_t most_sig_bits = 0`. The constructor is trivial as long as it is not user-defined (i.e. until the compiler can definitely prove that it is not running any user logic).

Reply:
If that doesn't work we should chose safety over convenience. We can write a templated serializer that works for all UUID_class types easily.

Reply:
> I think you can still initialize members as in int64_t most_sig_bits = 0. The constructor is trivial as long as it is not user-defined (i.e. until the compiler can definitely prove that it is not running any user logic).

That was one of the first things I tried.  The compiler thinks differently than our intuition :)

> If that doesn't work we should chose safety over convenience. We can write a templated serializer that works for all UUID_class types easily.

I'll take a stab at it.

If this works, We should be able to define utils::UUID itself as derived from UUID_class as well.
So all uuid-classes would be symmetrical.

Reply:
I think we should still keep using composition instead of inheritance, to avoid accidental up-casting.

Reply:
The idea is not to derive UUID_class<T> from UUID, but that rather define UUID itself as a strong class
like `struct UUID : public UUID_class<UUID>`, and that should be immune to accidental type casting to other types derived from `UUID_class` due to the template type tag.

Reply:
I don't really follow. How will inheriting `UUID` from some type make it a strong type?

Reply:
never mind.  I tried the common ancestor approach and it is too complicated.
Next version has a UUID_class<T> base template that carries a utils::UUID member.

The culprit requiring trivial types was in bit_cast, but the requirement was too harsh
as proven by https://en.cppreference.com/w/cpp/numeric/bit_cast requires only `std::is_trivially_copyable_v`.

I'm fixing that along the way.